torchvision/utils.py

```diff
@@ -395,33 +391,15 @@ def draw_keypoints(
     return torch.from_numpy(np.array(img_to_draw)).permute(2, 0, 1).to(dtype=torch.uint8)
 
 
-def _generate_color_palette(num_masks: int):
+def _generate_color_palette(num_masks: int, return_tensor: bool = True):
     palette = torch.tensor([2 ** 25 - 1, 2 ** 15 - 1, 2 ** 21 - 1])
-    return [tuple((i * palette) % 255) for i in range(num_masks)]
-
-
-def _generate_random_color() -> Tuple[int, int, int]:
-    """
-    Generates a random RGB Color.
-
-    Returns:
-        color Tuple(int, int, int): A Tuple containing Random RGB Color.
-    """
-    return tuple(torch.randint(256, (3,)).tolist())
-
-
-def _generate_random_color_palette(num_colors: int) -> List[Tuple[int, int, int]]:
-    """
-    Generates a random RGB Color palette.
-
-    Args:
-        num_colors (int): Integer denoting number of random RGB colors to generate
-
-    Returns:
-        colors(List[Tuple[int, int, int]): A List of Tuples each containing random RGB color.
-            Each color is Tuple containing RGB values.
-    """
-    return [_generate_random_color() for i in range(num_colors)]
+    clrs = []
+    for i in range(num_masks):
+        if return_tensor:
+            clrs.append((tuple((palette * i) % 255)))
+        else:
+            clrs.append(tuple(((palette * i) % 255).tolist()))
```

Comment:
Can you refactor the above 4 lines to avoid code duplication across the 2 cases? Try assigning to a temp var and doing `tolist` on it when necessary prior appending.

Reply:
Yeah can do :)